### PR TITLE
take thrift files as a parameter & arg1 matcher

### PR DESCRIPTION
**Pull request by ShanniLi**
r @kriskowal @Raynos

### CODE REVIEW

File: thrift/decoder.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        return simpleDecoder.decode(arg3);
+    }
+
+    var spec = specs[arg1s[0]];
+    if (!spec) {
+        return simpleDecoder.decode(arg3);
+    }
+
+    try {
+        var result = thriftify.fromBuffer(arg3, spec,
```
Comment (kriskowal) [inline thrift/decoder.js]: thrifty.fromBufferResult gives you a result object, so you don’t need a try/catch.

**Comment by kriskowal** [comment]
Looks alright otherwise.

File: thrift/decoder.js (hunk 1 above)
Comment (ShanniLi) [inline thrift/decoder.js]: It will throw if there is a version mismatch.

File: thrift/decoder.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    var files = fs.readdirSync(dir);
+    var fileRead = [];
+    files.forEach(function eachFile(file) {
+        var match = /([^\/]+)\.thrift$/.exec(file);
+        if (match) {
+            var spec = thriftify.readSpecSync(path.join(dir, file));
+            fileRead.push(match[0]);
+            for(var service in spec.servicesAndFunctions) {
```
Comment (Raynos) [inline thrift/decoder.js]: we do not loop like this; use `Object.keys(spec.servicesAndFunctions)`

**Comment by Raynos** [comment]
Otherwise lgtm;

File: thrift/decoder.js (hunk 2 above)
Comment (kriskowal) [inline thrift/decoder.js]: That’s right. `in` deoptimizes the function. Pattern is `var keys = Object.keys(object); for (var index = 0; index < keys.length; index++) { var key = keys[index]; /* ... */ }`

**Pull request merged**